### ISSUE TITLE
Array is partially allowed as a BigSource

### ISSUE THREAD

**Issue by ulexxander** [MikeMcl/big.js#163]
Big.js will not throw exception if I give it array containing one valid BigSource.

Which can be unsafe in some cases, especially because it is not specified in **typescript type definitions** (index.d.ts)

```ts
export type BigSource = number | string | Big;
```

Check this example:

```ts
import Big from 'big.js';

try {
  console.log('first case: one element in arr');
  console.log(Big(['1.2345'] as any).toString());
} catch (e) {
  console.log('err', e.message);
}

try {
  console.log('second case: two elements in arr');
  console.log(Big([5, 100] as any).toString());
} catch (e) {
  console.log('err', e.message);
}

try {
  console.log('third case: empty arr as source');
  console.log(Big([] as any).toString());
} catch (e) {
  console.log('err', e.message);
}
```

Output:
```
first case: one element in arr
1.2345
second case: two elements in arr
err [big.js] Invalid number     
third case: empty arr as source 
err [big.js] Invalid number
```

These are package versions from my `package.json`:
```json
{
  "@types/big.js": "^6.0.2",
  "big.js": "^6.0.3",
}
```

**Comment by ulexxander** [comment]
It will not work if I set `Big.strict` to `true`.

But a can not do that without casting Big to **any** in typescript, because this field is not specified in type definitions as well.

![image](https://user-images.githubusercontent.com/55105865/112154884-02762180-8be5-11eb-8665-0fc8fefdecf3.png)

Consider specifying "strict" field in types and adding tuple of primitive big source `(string | number | Big)` to current union type `BigSource`, so it will look something like this:

```ts
export type BigSourcePrimitive = number | string | Big;
export type BigSource = BigSourcePrimitive | [BigSourcePrimitive];
```

Thank you, I hope I explained everything well.
